Systemd running dind and stats fails

Reproducible on a systemd machine:

```
TESTFLAGS='-v -run TestGetContainerStats' make test-integration-cli
```

Probably something with systemd cgroups

btw, I see this on RHEL/CentOS 7 as well as ubuntu 14.04 for `make test-integration-cli`

    [PASSED]: version - verify that it works and that the output is properly formatted
    FAIL
    coverage: 71.0% of statements
    exit status 1
    FAIL    github.com/docker/docker/integration-cli    1319.723s
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon30809327/docker.pid
    +++ kill 27690
    /go/src/github.com/docker/docker/hack/make/.integration-daemon-stop: line 6: wait: pid 27690 is not a child of this shell
    warning: PID 27690 from /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon30809327/docker.pid had a nonzero exit code
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon32588737/docker.pid
    +++ kill 22034
    /go/src/github.com/docker/docker/hack/make/.integration-daemon-stop: line 6: wait: pid 22034 is not a child of this shell
    warning: PID 22034 from /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon32588737/docker.pid had a nonzero exit code
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon49502928/docker.pid
    +++ kill 22263
    /go/src/github.com/docker/docker/hack/make/.integration-daemon-stop: line 6: wait: pid 22263 is not a child of this shell
    warning: PID 22263 from /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/daemon49502928/docker.pid had a nonzero exit code
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/docker.pid
    +++ kill 685
    make: *** [test-integration-cli] Error 1

`TESTFLAGS='-v -run TestGetContainerStats' make test-integration-cli`

-works on ubuntu 14.04

-fails on CentOS 6 (not sure why it complains `no such file or dir`):


    Sending build context to Docker daemon 
    time="2015-02-18T03:02:57Z" level="fatal" msg="Post http:///var/run/docker.sock/v1.18/build?dockerfile=Dockerfile&rm=1&t=busybox: dial unix /var/run/docker.sock: no such file or directory" 
    +++ tar -cC /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/emptyfs .
    +++ docker load
    time="2015-02-18T03:02:57Z" level="fatal" msg="Post http:///var/run/docker.sock/v1.18/images/load: dial unix /var/run/docker.sock: no such file or directory" 
    + go test -v -run TestGetContainerStats -test.timeout=30m github.com/docker/docker/integration-cli
    === RUN TestGetContainerStats
    time="2015-02-18T03:03:04Z" level="fatal" msg="Get http:///var/run/docker.sock/v1.18/containers/json?all=1: dial unix /var/run/docker.sock: no such file or directory. Are you trying to connect to a TLS-enabled daemon without TLS?" 

    --- FAIL: TestGetContainerStats (0.12s)
    	docker_api_containers_test.go:302: Error on container creation: exit status 1, output: "time=\"2015-02-18T03:03:03Z\" level=\"fatal\" msg=\"Post http:///var/run/docker.sock/v1.18/containers/create?name=statscontainer: dial unix /var/run/docker.sock: no such file or directory. Are you trying to connect to a TLS-enabled daemon without TLS?\" \n"
    FAIL
    coverage: 4.5% of statements
    exit status 1
    FAIL	github.com/docker/docker/integration-cli	0.137s
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/docker.pid
    +++ kill 644
    /go/src/github.com/docker/docker/hack/make/.integration-daemon-stop: line 5: kill: (644) - No such process
    make: *** [test-integration-cli] Error 1

-fails on CentOS 7:

    + go test -v -run TestGetContainerStats -test.timeout=30m github.com/docker/docker/integration-cli
    === RUN TestGetContainerStats
    --- FAIL: TestGetContainerStats (4.39s)
    	docker_api_containers_test.go:334: EOF
    FAIL
    coverage: 11.2% of statements
    exit status 1
    FAIL	github.com/docker/docker/integration-cli	4.398s
    ++++ cat /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-integration-cli/docker.pid
    +++ kill 703
    make: *** [test-integration-cli] Error 1

--- a/cgroups/fs/apply_raw.go
+++ b/cgroups/fs/apply_raw.go
@@ -220,16 +220,16 @@ func getCgroupData(c *configs.Cgroup, pid int) (*data, error) {
 	}, nil
 }
 
-func (raw *data) parent(subsystem string) (string, error) {
+func (raw *data) parent(subsystem, mountpoint string) (string, error) {
 	initPath, err := cgroups.GetInitCgroupDir(subsystem)
 	if err != nil {
 		return "", err
 	}
-	return filepath.Join(raw.root, subsystem, initPath), nil
+	return filepath.Join(mountpoint, initPath), nil
 }
 
 func (raw *data) path(subsystem string) (string, error) {
-	_, err := cgroups.FindCgroupMountpoint(subsystem)
+	mnt, err := cgroups.FindCgroupMountpoint(subsystem)
 	// If we didn't mount the subsystem, there is no point we make the path.
 	if err != nil {
 		return "", err
@@ -240,7 +240,7 @@ func (raw *data) path(subsystem string) (string, error) {
 		return filepath.Join(raw.root, subsystem, raw.cgroup), nil
 	}
 
-	parent, err := raw.parent(subsystem)
+	parent, err := raw.parent(subsystem, mnt)
 	if err != nil {
 		return "", err
 	}

